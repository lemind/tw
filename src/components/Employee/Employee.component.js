import React from 'react';
import { Popover, OverlayTrigger } from 'react-bootstrap';

//import './Employee.styles.less';

export class EmployeeComponent extends React.Component {
  constructor(props) {
    super();

    this.select = props.onSelect;

    this.state = {
      employee: props.employee
    };

    this.handleClick = this.handleClick.bind(this);
  }

  getPopover(shortDescription) {
    return <Popover id="popover-trigger-hover-focus">
      { shortDescription }
    </Popover>
  };

  handleClick(event) {
    event.preventDefault();
    this.select(this.state.employee);
  };

  render() {
    const employee = this.state.employee;

    return (
      <div>
        <OverlayTrigger
          trigger={['hover', 'focus']}
          placement="bottom"
          overlay={this.getPopover(employee.shortDescription)}
        >
          <div
            className={"row " + (employee.selected ? "bg-success text-white" : "")}
            onClick={this.handleClick}
          >
            <span className="col-xs-5">{employee.firstName} {employee.lastName}</span>
            <span className="col-xs-3">{employee.role}</span>
            <span className="col-xs-4">{employee.experience}</span>
          </div>
        </OverlayTrigger>
      </div>
    )
  };
}