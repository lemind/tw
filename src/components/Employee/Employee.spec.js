import React from 'react';
import { shallow, mount } from 'enzyme';

import { EmployeeComponent as Employee } from './Employee.component';

it('should render Employee', () => {
  expect(shallow(<Employee employee={ {} } />)).toMatchSnapshot();
});

it('should call function on click Employee component', () => {
  const mockCallback = jest.fn();
  const component = mount(<Employee employee={ {} } onSelect={ mockCallback } />);
  component.find('.row').simulate('click');
  expect(mockCallback).toBeCalled()
});
