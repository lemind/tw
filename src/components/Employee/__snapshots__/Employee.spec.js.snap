// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Employee 1`] = `
<div>
  <OverlayTrigger
    defaultOverlayShown={false}
    overlay={
      <Popover
        bsClass="popover"
        id="popover-trigger-hover-focus"
        placement="right"
      />
    }
    placement="bottom"
    trigger={
      Array [
        "hover",
        "focus",
      ]
    }
  >
    <div
      className="row "
      onClick={[Function]}
    >
      <span
        className="col-xs-5"
      >
         
      </span>
      <span
        className="col-xs-3"
      />
      <span
        className="col-xs-4"
      />
    </div>
  </OverlayTrigger>
</div>
`;
